### PR TITLE
feat: nginx configurations

### PR DESCRIPTION
**Pull request by TorstenDittmann**
## What does this PR do?

- adds compression fallbacks, cache and security headers to nginx
- adds precompress to static build

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

✅

### CODE REVIEW

File: docker/nginx.conf [hunk 1]
```diff
@@ -1,12 +1,34 @@
...
+    gzip_static on;
+
+    # Deny IE browsers from going into quirks mode
+    add_header X-UA-Compatible "IE=Edge";
+    # X-Frame-Options is to prevent from clickJacking attack
+    add_header X-Frame-Options SAMEORIGIN;
+    # This header enables the Cross-site scripting (XSS) filter
+    add_header X-XSS-Protection "1; mode=block;";
+    # disable content-type sniffing on some browsers.
+    add_header X-Content-Type-Options nosniff;
```
Comment (stnguyen90) [inline docker/nginx.conf:23]: Shouldn't I see these in the response headers? 🧐 

image

File: docker/nginx.conf [hunk 2]
```diff
@@ -1,12 +1,34 @@
...
+
     location /console {
         root /usr/share/nginx/html;
         index index.html index.htm;
         try_files $uri /console/index.html;
+        expires $expires;
```
Comment (stnguyen90) [inline docker/nginx.conf:29]: I don't seem to see this in the response header

Comment (stnguyen90) [reply docker/nginx.conf:29]: maybe the Cache-Control one is causing a conflict?

**Review by stnguyen90 — COMMENTED** [review]
I tried browsing to http://127.0.0.1:3000/auth/oauth2/success but it redirected me to /console. Is that expected?

**Comment by TorstenDittmann** [comment]
> I tried browsing to http://127.0.0.1:3000/auth/oauth2/success but it redirected me to /console. Is that expected?

That redirect will be handled on the appwrite API side. Due to loadbalancer configurations, the console cannot act outside of the `/console` basepath. So in production it redirects to `http://127.0.0.1:3000/console/auth/oauth2/success`.

File: docker/nginx.conf (hunk 1 above)
Comment (TorstenDittmann) [reply docker/nginx.conf]: Good catch, looks like adding the cache headers kinda broke them. fixed 👍🏻

File: docker/nginx.conf (hunk 2 above)
Comment (TorstenDittmann) [reply docker/nginx.conf:22]: Only will be applied to non `text/html` requests 👍🏻

**Review by stnguyen90 — APPROVED** [review]

**Pull request merged**